### ISSUE TITLE
New rotations create logs with a year of 1970

### ISSUE THREAD

**Issue by mstenta** [farmOS/farm_grazing#14]
Noticed this today: when rotations are initially created in a plan, the logs that are created for them have incorrect dates. They have a year of 1970.

Notably, this can be fixed automatically by clicking the "Save" button on the rotations form, which automatically updates all the logs.

I need to test this again to replicate it and make sure I understand when/how it's happening. I noticed it as Jess at Paicines Ranch was using it for the first time today.

**Comment by mstenta** [comment]
I figured out why this was happening. In the submit function for creating rotations, it was looking up the date of the last rotation to use as a start date when you add a new rotation. However, when adding the FIRST rotation, there are no other rotations to refer to, so it was ending up with a starting timestamp of zero, and all rotations after that were just adding to that. I fixed it by checking to see if a previous rotation was found.  If not, it uses the start date of the plan.
